.dev/docker/windows/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# escape=`
+
+FROM mcr.microsoft.com/windows/servercore:ltsc2019
+
+# Use "--build-arg platform=x64" for 64 bit or x86 for 32 bit.
+ARG platform
+
+# Download channel for fixed install.
+ARG CHANNEL_URL=https://aka.ms/vs/16/release/channel
+ADD ${CHANNEL_URL} C:\TEMP\VisualStudio.chman
+
+# Download and install Build Tools for Visual Studio 2019 for native desktop workload.
+ADD https://aka.ms/vs/16/release/vs_buildtools.exe C:\TEMP\vs_buildtools.exe
+
+RUN C:\TEMP\vs_buildtools.exe --quiet --wait --norestart --nocache `
+    --channelUri C:\TEMP\VisualStudio.chman `
+    --installChannelUri C:\TEMP\VisualStudio.chman `
+    --add Microsoft.VisualStudio.Workload.VCTools --includeRecommended`
+    --installPath C:\BuildTools
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+RUN iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+RUN choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System' -y --no-progress
+
+# Install vcpkg
+ADD https://github.com/Microsoft/vcpkg/archive/master.zip C:\TEMP\vcpkg-master.zip
+
+RUN $ErrorActionPreference = 'Stop';                                  `
+    Expand-Archive -Path C:\TEMP\vcpkg-master.zip -DestinationPath .; `
+    rm C:\TEMP\vcpkg-master.zip;                                      `
+    cd .\vcpkg-master;                                                `
+    .\bootstrap-vcpkg.bat;                                            `
+    .\vcpkg integrate install
+    
+RUN .\vcpkg-master\vcpkg install lz4 --triplet $Env:platform-windows
```

Comment:
Any reason for separate `RUN` commands? You can also use env variable to set [the default triplet](https://vcpkg.readthedocs.io/en/latest/users/triplets/#additional-remarks) once for all vcpkg command.

Also, it is customary in Dockerfiles to build, install and remove the cache/build-files in one step to keep the container size low.

Reply:
Simple to have each install cached, if it failed during some of all of the installs.. timouts for download etc. I wouldn't have to start all over.
It failed miserably often once, but maybe I was just unlucky.

Is the cache/build files part of the container? Thought those would just be locally...

Reply:
Yes. Sadly the cache is part of the container. For debugging and creating, it's fine to have separate `RUN` stages (as you said, faster iterations), but for a production-ready container, the flatter the images is, the better it is (Image size will reduce significantly, and since the container is known to build properly, that's not a worry)